### PR TITLE
added null check for weather key

### LINKED ISSUE #70: Weather module does not check for null API key
**Describe the bug**
Weather module does not check for a null key. Results in widget constantly trying to parse a null value and throwing errors.

**Expected behavior**
Weather module does not load on null value

**Desktop (please complete the following information):**
 - Distribution: Arch
 - Window Manager/Desktop Environment: Hyprland

### PR DESCRIPTION
**Pull request by matavach**
fixes #70 

I see two approaches for solving this, null checking on the weather module itself or null checking in the index.ts for the calendar module.

I chose the former, since it would keep the fix if we decide to use the weather module somewhere else in the code.

I added a visible property to the top level box, because otherwise it would render an empty box.

image

with the visible property also doing a null check:
image

Do we want to include a warning for users if weather is disabled?

### CODE REVIEW

**Comment by Jas-SinghFSU** [comment]
Hmmm, is the calendar not displaying default values? My original idea was to leave it in there and display 0s for the weather values until someone put in the weatherAPI key. That way, the user would know that the Weather module exists upon first glance and they can activate it with a key.

**Comment by matavach** [comment]
It does display zeroes, however, as is, it will poll constantly and throw errors nonstop.
I will adjust this to display the 0s without polling.

**Comment by matavach** [comment]
With the new commit, this is the view with no api key available.
image
It also stops polling until a key is presented.

File: modules/menus/calendar/weather/index.ts [hunk 1]
```diff
@@ -19,6 +19,9 @@ const WeatherWidget = () => {
                 Utils.merge(
                     [key.bind("value"), interval.bind("value"), location.bind("value")],
                     (weatherKey, weatherInterval, loc) => {
+                        if(!weatherKey){
```
Comment (Jas-SinghFSU) [inline modules/menus/calendar/weather/index.ts:22]:
```suggestion
                        if(!weatherKey) {
```

**Pull request merged**